tools/dev-doctor/main.go

```diff
@@ -0,0 +1,121 @@
+// Command doctor checks the development setup for common problems.
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"text/tabwriter"
+
+	"github.com/blang/semver"
+)
+
+// A checkResult is the result of a check.
+type checkResult int
+
+const (
+	checkSkipped checkResult = -1 // The check was skipped.
+	checkOK      checkResult = 0  // The check completed and did not find any problems.
+	checkWarning checkResult = 1  // The check completed and found something that might indicate a problem.
+	checkError   checkResult = 2  // The check completed and found a problem.
+	checkFailed  checkResult = 3  // The check could not be completed.
+)
+
+// A check is an individual check.
+type check interface {
+	id() string
+	run() (checkResult, string)
+}
+
+var checkResultStr = map[checkResult]string{
+	checkSkipped: "skipped",
+	checkOK:      "ok",
+	checkWarning: "warning",
+	checkError:   "error",
+	checkFailed:  "failed",
+}
+
+func main() {
+	checks := []check{
+		osArchCheck{},
+		&binaryCheck{
+			name:          "go",
+			versionArgs:   []string{"version"},
+			versionRegexp: regexp.MustCompile(`go version go(\d+\.\d+\.\d+)`),
```

Comment:
I don't have a patch version number to my output. I think we can safely ignore that last digit anyway.
```
$ go version
go version go1.14 linux/amd64
```

Reply:
Fixed, thanks.